Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,27 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+# Truncates a text string to fit within a byte size constraint,
+# preserving character encoding validity. The returned string will
+# be no longer than the specified max_bytes, though the exact
+# shortfall may vary.
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
```

Comment:
`brew test-bot` truncates step output by chopping off the beginning of the file, effectively doing `front_weight = 0`. If this has support for it, `test-bot` could be refactored to share this code instead of having a separate text-aware truncation implementation.